presto-main/src/main/java/com/facebook/presto/execution/TaskState.java

```diff
@@ -13,11 +13,15 @@
  */
 package com.facebook.presto.execution;
 
+import com.facebook.drift.annotations.ThriftEnum;
+import com.facebook.drift.annotations.ThriftEnumValue;
+
 import java.util.Set;
 import java.util.stream.Stream;
 
 import static com.google.common.collect.ImmutableSet.toImmutableSet;
 
+@ThriftEnum
 public enum TaskState
```

Comment:
I would add a comment saying that the order needs to be backwards compatible, because in theory the TaskState could be persisted.

Reply:
Added explicit codes for enums